openpype/hosts/nuke/api/plugin.py

```diff
@@ -818,15 +817,14 @@ def __init__(self,
 
         self.file = self.fhead + self.name + ".{}".format(self.ext)
         if ".{}".format(self.ext) not in VIDEO_EXTENSIONS:
-            # filename would be with frame hashes if
-            # the file extension is not in video format
-            filename = get_head_filename_without_hashes(
-                self.path_in, self.name)
-            self.file = filename
+            filename = os.path.basename(self.path_in)
+            self.file = self.fhead + self.name + ".{}".format(
+                filename.split(".", 1)[-1]
+            )
```

Comment:
This would still fail with files that contain a `dot` before the hashes or frame numbers.

```
hello.world.####.exr
hello.world.0001.exr
```
Would now both make:
```
hello.world.{name}.world
```

Also, with this logic we now know for a fact that `self.file` will not have the extension since we creating the filename ourselves without the extension

so the logic below would actually be splitting of `.world` and then replace that with the extension so we end up with `hello.world.{name}.{ext}` which does not have the frame number or hashes which would not work for image based outputs?

Reply:
maybe we can try to use `replace `here for the files(although it is not likeable)
`self.file =self.fname.replace(self.fhead, self.fhead+self.name)`